### PR TITLE
Make ExodusViewer side widgets scrollable

### PR DESCRIPTION
**Pull request by brianmoose**
Even with #8855, on lower resolution displays the widgets could get messed up when expanding widgets. Expanding widgets also increased the window size which could cause part of the peacock window to be inaccessible.
This turns the whole area into a QSrollArea so that if the window is resized or the widgets need more room, the area becomes scrollable.

Also set a PluginManager variable on each plugin. This is not a very good thing to do since ideally each plugin would have no need to know its PluginManager or other plugins but there are a couple cases where this is extremely helpful.

### CODE REVIEW

File: python/peacock/ExodusViewer/ExodusViewer.py [hunk 1]
```diff
@@ -23,9 +23,9 @@ class ExodusViewer(peacock.base.ViewerBase):
     def commandLineArgs(parser):
         parser.add_argument('--exodus', '-r', nargs='*', default=[], help="A list of ExodusII files to open.")
 
-    def __init__(self, plugins=[VTKWindowPlugin, FilePlugin, GoldDiffPlugin, VariablePlugin, \
+    def __init__(self, plugins=[FilePlugin, GoldDiffPlugin, VariablePlugin, \
                                 MeshPlugin, BackgroundPlugin, ClipPlugin, ContourPlugin, CameraPlugin, \
-                                OutputPlugin, MediaControlPlugin,
+                                OutputPlugin, VTKWindowPlugin, MediaControlPlugin,
```
Comment (brianmoose) [inline python/peacock/ExodusViewer/ExodusViewer.py]: I have no idea why this change is necessary but it is. With the original order the scroll area gets all screwed up when scrolling.

**Comment by brianmoose** [comment]
@aeslaughter See if you like the behavior and if it looks alright...

**Review by aeslaughter — APPROVED** [review]

**Comment by brianmoose** [comment]
The linux test failure wasn't due to this PR. The machine just got updated and apparently the ~/.ICEauthority file was causing problems.
Does this close #8826?

**Comment by permcody** [comment]
I don't think I like the idea of scrollbars in the window. It makes it feel cluttered. If we have to go to this, fine, but somehow we managed without them in the old Peacock. Is there any way to squeeze out a little more space? I guess I don't know how much space we're short on low resolution displays.

**Comment by brianmoose** [comment]
@aeslaughter I removed the frame and the horizontal "wiggle" that you showed me. You may want to confirm.

**Comment by friedmud** [comment]
Wait?  Are we remaking Paraview at this point?  One of the things I hate the most about Paraview is the freaking scroll area on the left.

If the widgets can't fit in 1024x768 (which is what a lot of projectors still are) then it's over-designed and too many widgets are showing.

**Comment by permcody** [comment]
This was my thought too. After looking at the interfaces though, I don't think what we have is bad. It's one scrollable interface that fights on modern displays. It only needs to scroll if you have a very small display. The feature set is still only a fraction of Paraview but slightly larger than the old Peacock. Take a look when you get a chance.

**Comment by aeslaughter** [comment]
It should only scroll if it is needed, which for me was on the lowest
resolution on my laptop only (1024x680). If you want less widgets we can
remove some to make it happen even less. However, adding this scrolling
should be included because as we progress people (e.g., Mastodon and Bison
teams are already is wanting to add features) will be adding plugins, which
when added may require things to scroll, for them.

**Comment by friedmud** [comment]
Sounds good.
